.github/workflows/rust.yml

```diff
@@ -91,3 +91,74 @@ jobs:
     steps:
     - uses: actions/checkout@v4
     - uses: EmbarkStudios/cargo-deny-action@v1
+
+  test-android:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        include:
+          - target: x86_64-linux-android
+            emulator-arch: x86_64
+          - target: i686-linux-android
+            emulator-arch: x86
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    - name: Install JDK
+      uses: actions/setup-java@v3
+      with:
+        distribution: 'zulu'
+        java-version: '21'
+
+    - name: Install Android SDK
+      uses: android-actions/setup-android@v2
+
+    - name: Install Android NDK
+      run: sdkmanager --install "ndk;25.2.9519653"
+
+    - name: Install Rust
+      uses: dtolnay/rust-toolchain@stable
+      with:
+        toolchain: stable
+        target: ${{ matrix.target }}
+
+    - uses: Swatinem/rust-cache@v2
+
+    - name: Install cargo-ndk
+      run: cargo install cargo-ndk
+
+    - name: Build unit tests for Android
+      run: cargo ndk -t ${{ matrix.target }} test --no-run
+
+    - name: Enable KVM group perms
+      run: |
+        echo 'KERNEL=="kvm", GROUP="kvm", MODE="0666", OPTIONS+="static_node=kvm"' | sudo tee /etc/udev/rules.d/99-kvm4all.rules
+        sudo udevadm control --reload-rules
+        sudo udevadm trigger --name-match=kvm
+
+    - name: Create emulator script
+      run: |
+        echo '#!/bin/bash
```

Comment:
Don't love having this longish shell script inside the YAML. Maybe extract it into a separate file?

Reply:
Extracted into a separate file. Mind taking another look?

https://github.com/quinn-rs/quinn/pull/1950/commits/3fa87f2a2ae803b6b4d13a3314cd1997395dbd16